src/main/java/com/synopsys/integration/detect/lifecycle/autonomous/AutonomousManager.java

```diff
@@ -120,4 +128,79 @@ public SortedMap<String, String> getAllScanSettingsProperties() {
 
         return scanSettingsProperties;
     }
+
+    public void updateScanSettingsProperties(SortedMap<String, String> propertiesMap, List<String> adoptedScanTypes) {
+        userProvidedProperties.forEach((propertyKey, propertyValue) -> {
+            Optional<String> scanTypeValue = findScanType(adoptedScanTypes, propertyKey);
+            determinePropertyTypeAndUpdate(propertyKey, propertyValue, scanTypeValue, true);
+        });
+        propertiesMap.forEach((propertyKey, propertyValue) -> {
+            Optional<String> scanTypeValue = findScanType(adoptedScanTypes, propertyKey);
+            determinePropertyTypeAndUpdate(propertyKey, propertyValue, scanTypeValue, false);
+        });
+
+        scanSettings.setDetectorSharedProperties(detectorSharedProperties);
+        scanSettings.setGlobalDetectProperties(globalProperties);
+    }
```

Comment:
This is a general function that updates properties in the scan settings file, userProvidedProperties are the properties which are provided by the user. Default Properties Map is the map that we got for default decisions.